src/rf/apps/workers/process.py

```diff
@@ -292,26 +294,30 @@ def check_timeout(self, record):
             return False
 
         try:
-            int(layer_id)
+            layer_id = int(layer_id)
             float(timeout)
         except ValueError:
             return False
 
+        try:
+            layer = Layer.objects.get(id=layer_id)
+        except Layer.DoesNotExist:
+            return True
+
+        log.debug('Timeout for layer %d', layer_id)
+
+        if layer.status in (enums.STATUS_COMPLETED, enums.STATUS_FAILED):
+            log.debug('Ending timeout')
```

Comment:
Consider making this more explicit:
log.debug('Ending timeout. Job is complete.')